libcudacxx/include/cuda/std/__atomic/functions/cuda_ptx_derived.h

```diff
@@ -339,6 +339,98 @@ _CCCL_DEVICE _Tp __atomic_exchange_n_cuda(_Tp volatile* __ptr, _Tp __val, int __
   return __ret;
 }
 
+template <typename _Tp, typename _Up, typename _Sco, __atomic_enable_if_not_native_arithmetic<_Tp> = 0>
+_CCCL_DEVICE _Tp __atomic_fetch_add_cuda(_Tp* __ptr, _Up __val, int __memorder, _Sco)
+{
+  return __atomic_fetch_update_cuda(
+    __ptr,
+    [__val](_Tp __old) {
+      return __old + __val;
+    },
```

Comment:
This will instantiate a struct everytime this function is instantiated, which is going to be quite often.

We need to pull those lambdas out into a struct that has the right operators

Reply:
Done, though you might want to see if my plan has any issues. I don't think it will be much improved.